design/accepted/kv_migration.md

```diff
@@ -0,0 +1,35 @@
+# KV Migration (export import phase 1) Design
+
+## Definitions
+1. KV Migration - The transition from schema based postgres implementation to key value store based postgres implementation
+2. DB Migration - The process of migrating postgres data using the postgres versioning schema
+3. KV Migration Version - “Schema” version for which the KV Migration will occur
+
+## Abstract
+Current migration mechanism relies on postgres versioning schema.
+KV Migration flow is a transitory mechanism, which will be needed only when upgrading lakeFS to the KV Migration Version. For efficiency and re-use we would like to create a preliminary version of the export-import feature to implement the KV Migration flow.
+
+## Considerations
+Although we cannot use the DB Migration flow to handle the KV migration, it must be incorporated into it and executed in the context of the DB migration
+DB migration must be performed up to the latest version before the KV Migration Version and only then KV Migration can happen
+KV Migration Version will be a consecutive version succeeding the last DB Migration version.
```

Comment:
But KV is an ongoing task, which is hidden behind a feature-flag. How would DB migration continue to work? How would we make KV migrations non-mandatory?

Reply:
Adding the information to the document